### PR TITLE
fix: No log file

### PR DESCRIPTION
**Pull request by 18202781743**
The log file will not be generated even though we call registerFileAppender when $HOME is empty, it is the default for Qt, But we can fallback it to get the home directory from its user. e.g. for dde-dconfig-daemon, it may be called by a specific user with a home directory but $HOME unset.

### CODE REVIEW

**Comment by asterwyx** [comment]
> Without log file
no log file

**Comment by asterwyx** [comment]
> It's not exist log file even though we call `registerFileAppender` when `$HOME` is empty, it is default for Qt, But we can fallback it to get homedir from it's user. e.g: for `dde-dconfig-daemon`, it may be called by a specific user, which has home directory but not have $HOME.

The log file will not be generated even though we call `registerFileAppender` when `$HOME` is empty, it is the default for Qt, But we can fallback it to get the home directory from its user. e.g. for `dde-dconfig-daemon`, it may be called by a specific user with a home directory but $HOME unset.

File: src/log/LogManager.cpp [hunk 1]
```diff
@@ -7,9 +7,25 @@
 #include <ConsoleAppender.h>
 #include <RollingFileAppender.h>
 #include <JournalAppender.h>
+#include "DStandardPaths"
```
Comment (asterwyx) [inline src/log/LogManager.cpp:10]:
```suggestion
#include <DStandardPaths>
```

File: src/log/LogManager.cpp [hunk 2]
```diff
@@ -86,14 +102,14 @@ QString DLogManager::getlogFilePath()
     //No longer set the default log path (and mkdir) when constructing now, instead set the default value if it's empty when getlogFilePath is called.
     //Fix the problem that the log file is still created in the default path when the log path is set manually.
     if (DLogManager::instance()->m_logPath.isEmpty()) {
-        if (QDir::homePath() == QDir::rootPath()) {
-            qWarning() << "unable to locate the cache directory."
-                       << "logfile path is empty, the log will not be written.\r\n"
-                       << (qgetenv("HOME").isEmpty() ? "the HOME environment variable not set" : "");
+        if (DStandardPaths::homePath().isEmpty()) {
+            qWarning() << "Unable to locate the cache directory, which has not home directory, and the log will not be written.";
```
Comment (asterwyx) [inline src/log/LogManager.cpp:106]:
```suggestion
            qWarning() << "Unable to locate the cache directory, cannot acquire home directory, and the log will not be written to file.";
```

**Review by kegechen — APPROVED** [review]

**Pull request merged**